TestFoundation/TestNSStream.swift

```diff
@@ -16,34 +16,43 @@
 #endif
 
 class TestNSStream : XCTestCase {
-    static var allTests: [(String, (TestNSStream) -> () throws -> Void)] {
+    typealias NSStreamTestCases = (String, (TestNSStream) -> () throws -> Void)
```

Comment:
this is tests for "is-a" value type wrapper a lot of the tests overlaps the "has-a" implementation
opinions on weather they are needed are welcome. whether to keep both or get rid of some from this set or the "has-a" set